src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -119,13 +119,19 @@ public int Count
                     {
                         return new KeyValuePair<string, object>("RequestPath", _httpContext.Request.Path.ToString());
                     }
-                    throw new IndexOutOfRangeException(nameof(index));
+                    else if (index == 2)
+                    {
+                        return new KeyValuePair<string, object>("CorrelationId", _correlationId);
```

Comment:
If the correlation id in the logs is going to tie requests that travel through any number of services in series, then it needs to be set even at the origin service.

It seems like the `CorrelationId` should be set to `_httpContext.TraceIdentifier` if `_correlationId` is `null`.

Reply:
No, this never overwrites the request id. That's not right.

Reply:
I think then that this `CorrelationId` is really a mis-labeled `ParentRequestId` and that an actual `CorrelationId` should be added to reflect the root request id shared across all services that come after the root service that is handling a request made by a user.

This is my understanding of the spec at [FlatRequestId.md](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/FlatRequestId.md#non-hierarchical-request-id-example).  In that documentation, the correlation id is set in Service A and remains the same in all chained services.

I opened a new issue #1384 to this effect.